ios/RNCardConnectReactLibrary.m

```diff
@@ -37,15 +103,216 @@ - (dispatch_queue_t)methodQueue
+        case BMSSwiperConnectionStateConfiguring:
+
+            [self debug:@"connection state changed: configuring"];
+
+            // TODO: is this right, does it fire??
+            [self sendEventWithName:@"BoltOnDeviceBusy" body:@{}];
+            break;
+        case BMSSwiperConnectionStateConnecting:
+
+            [self debug:@"connection state changed: connecting"];
+
+            [self sendEventWithName:@"BoltOnSwiperConnecting" body:@{}];
+            break;
+        case BMSSwiperConnectionStateSearching:
+            [self debug:@"connection state changed: searching"];
+            // ignore for now?
+            break;
+        default:
+            break;
+    }
+}
+
+- (void)swiperDidStartCardRead:(BMSSwiper *)swiper {
+
+    [self sendEventWithName:@"BoltOnTokenGenerationStart" body:@{}];
+}
+
+- (void)swiper:(BMSSwiper *)swiper didGenerateTokenWithAccount:(BMSAccount *)account completion:(void (^)(void))completion {
+
+    if (account.name) {
+        [self debug:account.name];
+        [self sendEventWithName:@"BoltOnTokenGenerated" body:@{
+            @"token": account.token,
+            @"name": account.name
+        }];
+    }
+    // NFC taps have not been providing a name
+    // TODO: how to just mutate an object to send with the event?
+    else {
+        [self sendEventWithName:@"BoltOnTokenGenerated" body:@{
+            @"token": account.token
+        }];
+    }
+
+    // TODO: pass along the expiration date if we have it
+    // if (account.expirationDate) {
+    //     NSDateFormatter *formatter = [[NSDateFormatter alloc] init];
+    //     [formatter setDateFormat:@"MMyy"];
+    //     NSString *stringFromDate = [formatter stringFromDate:account.expirationDate];
+    // }
+
+    // store the completion block, so we can reactivate the device when needed
+    self.restartReaderBlock = completion;
+}
+
+- (void)swiper:(BMSSwiper *)swiper didFailWithError:(NSError *)error completion:(void (^)(void))completion {
+
+    // store the completion block, so we can reactivate the device when needed
+    self.restartReaderBlock = completion;
+
+    // this error will get issued repeatedly while connecting, ignore
+    if (self.isConnecting && [error.localizedDescription isEqualToString:@"Failed to connect to device."]) {
+        // this is where the error for failing to find the serial number surfaces
+        if (error.userInfo) {
+            [self debug:@"user info array"];
+            for(id key in error.userInfo) {
+                NSString* userInfoString = [error.userInfo objectForKey:key];
+                [self debug:userInfoString];
+            }
+        }
+        else {
+            [self debug:@"No user info array"];
+        }
+        
+        return;
+    }
+
+    // the device will automatically activate after connecting
+    // we cancel this transaction to complete the connection process
+    if (self.isConnecting && [error.localizedDescription isEqualToString:@"Canceled transaction."]) {
```

Comment:
Some more special strings down here that it might be nice to make constants.  Not too hung-up on it though, I don't think it would need to hold-up this PR.